internal/rbd/rbd_util.go

```diff
@@ -993,6 +993,9 @@ func genVolFromVolumeOptions(ctx context.Context, volOptions, credentials map[st
 }
 
 func (rv *rbdVolume) validateImageFeatures(imageFeatures string) error {
+	if strings.TrimSpace(imageFeatures) == "" {
```

Comment:
@mnaser I have seen some issues opened failed to mount the PVC when the imageFeatures are not set in storageclass https://github.com/ceph/ceph-csi/issues/1217#issuecomment-655871773 https://github.com/ceph/ceph-csi/issues/1293#issuecomment-665752463 etc. IMO below changes to fix the backward compatibility issue and also to provide a proper error message to the user during CreateVolume if the imageFeature is not set.

* Return proper error message from `validateImageFeatures` function which will get returned to the user in CreateVolume
* In NodestageVolume to support backward compatibility we can discard if `validateImageFeatures` returns the above error message and continue for the next operation.

Reply:
alternativity PR https://github.com/ceph/ceph-csi/pull/2139 to fix this completely

Reply:
> * Return proper error message from `validateImageFeatures` function which will get returned to the user in CreateVolume

This error message will be wrapped as following and maybe misleading 
https://github.com/ceph/ceph-csi/blob/7cbad9305f71b70040509ed854f1d65c7c624d4f/internal/rbd/rbd_util.go#L979-L982
> * In NodestageVolume to support backward compatibility we can discard if `validateImageFeatures` returns the above error message and continue for the next operation.

By discarding error, we would fail to enforce required `imageFeatures` option on static PVs(whose attributes can be edited). 

---
IMO, we need to validate and enforce required parameter like `imageFeatures` in https://github.com/ceph/ceph-csi/blob/7cbad9305f71b70040509ed854f1d65c7c624d4f/internal/rbd/controllerserver.go#L94 
by adding something like 
```go
	if _, ok := req.GetParameters()["imageFeatures"]; !ok {
		return nil, status.Error(codes.InvalidArgument, "missing required parameter imageFeatures")
	}
````
and in https://github.com/ceph/ceph-csi/blob/7cbad9305f71b70040509ed854f1d65c7c624d4f/internal/rbd/nodeserver.go#L115
Allow non-static volume to maintain backwards compatibility while enforcing the paramter for static pvs.
(newly provisioned non-static volume can be let go, since we enforce it during createrequest)
```go
	if _, ok := req.GetParameters()["imageFeatures"]; !ok && !staticVol {
		return nil, status.Error(codes.InvalidArgument, "missing required parameter imageFeatures")
	}
````
cc @nixpanic @humblec @Madhu-1

Reply:
I think this would still leave use cases like me that have PV's which I cannot edit with no image features that refuse to be mounted.